common/membership/ringpop/hostinfo.go

```diff
@@ -58,7 +70,29 @@ func (hi *hostInfo) Label(key string) (string, bool) {
 	return value, ok
 }
 
-// SetLabel sets the label.
-func (hi *hostInfo) SetLabel(key string, value string) {
-	hi.labels[key] = value
+// summary returns a shorthand summary string suitable for logging.
+func (hi *hostInfo) summary() string {
+	s := hi.GetAddress()
+	for k, v := range hi.labels {
+		switch k {
+		case roleKey, portKey:
+			// skip these, they can be determined from context
+		default:
+			s += fmt.Sprintf("+%s=%s", k, v)
+		}
+	}
+	return s
+}
+
+// checksumLabels returns a checksum of a labels map
+func checksumLabels(labels map[string]string) uint64 {
+	var c uint64
+	for k, v := range labels {
+		kfp := farm.Fingerprint64([]byte(k))
+		vfp := farm.Fingerprint64([]byte(v))
+		// use xor to combine different labels so that it comes out the same with any iteration
+		// order, without needing to sort.
+		c ^= 5*kfp + 7*vfp
```

Comment:
Out of curiosity why 5 and 7 specifically?

Reply:
they're small and relatively prime, so that swapping a key and value (however weird that would be) is sure to change the checksum. actually I wanted to do a bitwise rotate and thought that Go didn't have that in the stdlib, but it turns out it does. changed.